### PR TITLE
Fix up Dockerfile to use ubuntu:bionic and actually build

### LINKED ISSUE #3539: Dockerfile does not build
<b>System Information:</b>

AWS EC2 t2.micro instance running a free Ubuntu 18.04 amd64 image

Using `docker-ce` from Docker repos

__________________________________________________________________________________________

Quite simply, the Docker build of Cockatrice (required for the Docker-based `servertrice` setup) does not work. The main issue is that not all necessary files transferred into the Docker container (`.git` is missing, and is needed by cmake). However, even doing this transfer manually leads to further issues.

I have a PR pending for fixing this, and it will be attached soon.

### PR DESCRIPTION
**Pull request by fsufitch**
## Related Ticket(s)
- Fixes #3539

## Short roundup of the initial problem

The servertrice Docker build does not work.

## What will change with this Pull Request?
- `Dockerfile` is revamped to use `ubuntu:bionic`, no PPAs, and transfer all the correct files to the container.
- Building the image specified by the `Dockerfile` now works.

## Screenshots
N/A

### CODE REVIEW

**Comment by ZeldaZach** [comment]
@ebbit1q Can you test this please?

**Comment by fsufitch** [comment]
Tested and runs on an AWS Ubuntu Bionic amd64 image, and on a Fedora 29 Workstation laptop.

Also, updated the Docker instructions on the wiki to follow a process that actually works: https://github.com/Cockatrice/Cockatrice/wiki/Setting-up-Servatrice#using-docker

**Comment by ebbit1q** [comment]
Looks fine, will test later.

File: Dockerfile [hunk 1]
```diff
@@ -1,9 +1,8 @@
-FROM ubuntu:trusty
+FROM ubuntu:bionic
 MAINTAINER Zach Halpern <zahalpern+github@gmail.com>
 
 RUN apt-get update && apt-get install -y software-properties-common
-RUN apt-add-repository ppa:ubuntu-sdk-team/ppa
-RUN add-apt-repository -y ppa:smspillaz/cmake-master
+#RUN add-apt-repository -y ppa:smspillaz/cmake-master
```
Comment (tooomm) [inline Dockerfile]: I think we can remove that too as 18.04 LTS finally includs a more recent cmake (3.10), see https://packages.ubuntu.com/bionic/cmake

The old one only had 2.8 and we require [3.1](https://github.com/Cockatrice/Cockatrice/blob/master/CMakeLists.txt#L9)

Comment (fsufitch) [reply Dockerfile]: Removed that line.

File: Dockerfile [hunk 2]
```diff
@@ -1,30 +1,29 @@
...
   qttools5-dev\
-  libqt5sql5-mysql
+  software-properties-common
 
-ENV dir /home/servatrice/code
-WORKDIR $dir
+COPY . /home/servatrice/code/
+WORKDIR /home/servatrice/code
+
+<<<<<<< HEAD
```
Comment (ZeldaZach) [inline Dockerfile]: Nice try

Comment (ebbit1q) [reply Dockerfile]: WTF

File: Dockerfile [hunk 3]
```diff
@@ -1,30 +1,23 @@
...
+FROM ubuntu:bionic
 MAINTAINER Zach Halpern <zahalpern+github@gmail.com>
 
-RUN apt-get update && apt-get install -y software-properties-common
-RUN apt-add-repository ppa:ubuntu-sdk-team/ppa
-RUN add-apt-repository -y ppa:smspillaz/cmake-master
 RUN apt-get update && apt-get install -y\
-  build-essential g++\
+  build-essential\
   cmake\
```
Comment (tooomm) [inline Dockerfile]: Do we still need this?
I think we use the bundled cmake now?

Comment (ebbit1q) [reply Dockerfile]: We still need to actually install cmake to use cmake.

**Pull request merged**